pkg/reports/store.go

```diff
@@ -70,3 +93,32 @@ func (s *Store) Read(ctx context.Context, workload kube.Object) (vulnerabilities
 	}
 	return reports, nil
 }
+
+func (s *Store) getRuntimeObjectFor(ctx context.Context, workload kube.Object) (metav1.Object, error) {
```

Comment:
I'm not sure about this but I think you don't actually need the "real" type just to set it as owner, presumably you need just the metadata, so instead of this type switch you can just get it into a PartialObjectMetadata?

Reply:
That's a good point @itaysk . I actually tried that without type switch using PartialObjectMetadata, but apparently the client instantiated by the operator manager requires type information, i.e. I got the following error:

```
no kind is registered for the type v1.PartialObjectMetadata
```